src/CartAbandonmentService.php

```
@@ -0,0 +1,571 @@
+   * @throws \InvalidArgumentException
+   *   When the check interval is outside the valid range.
+   */
+  private function validateCheckInterval(int $check_interval): void {
+    if ($check_interval < self::MIN_CHECK_INTERVAL_MINUTES || $check_interval > self::MAX_CHECK_INTERVAL_MINUTES) {
+      throw new \InvalidArgumentException(sprintf(
+        'Check interval must be between %d and %d minutes',
+        self::MIN_CHECK_INTERVAL_MINUTES,
+        self::MAX_CHECK_INTERVAL_MINUTES
+      ));
+    }
+  }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'validateCheckInterval'. (Unused Code Rules)

(UnusedPrivateMethod)